python/shotgun_globals/cached_schema.py

```diff
@@ -738,43 +738,63 @@ def get_status_color(cls, status_code, project_id=None):
         return status_color
 
     @classmethod
-    def field_is_editable(cls, sg_entity_type, field_name):
+    def field_is_editable(cls, sg_entity_type, field_name, project_id=None):
         """
         Returns a boolean identifying the editability of the entity's field.
 
         :param str sg_entity_type: the entity type
         :param str field_name: the field name to check editibility
+        :param project_id:  The id of the project.
+                            If None, the current context's project will be used.
```

Comment:
hmm, wondering if this is the right logic? Historically, the logic was such that there was a global schema that worked everywhere. Then we added per-project support as an optional setting - see the docs for the shotgun API:

https://github.com/shotgunsoftware/python-api/blob/master/shotgun_api3/shotgun.py#L1213

I think it probably makes more sense to be consistent with the underlying Shotgun API that we are caching here and interpret `project_id=None` as "give me the site-wide setting" rather than "give me the current project setting". Calling code can easily determine the context for the current project.

Generally, we are moving in a direction where we will be working increasingly in a site context (because of TaaP) and should be mindful about the case where we don't have a current project.